src/utils/builtin_metrics.h

```diff
@@ -15,19 +15,35 @@
 // specific language governing permissions and limitations
 // under the License.
 
-#include "perf_counter_wrapper.h"
+#pragma once
+
+#include <memory>
+
+#include "utils/metrics.h"
+#include "utils/ports.h"
 #include "utils/singleton.h"
 
 namespace dsn {
-class builtin_counters : public dsn::utils::singleton<builtin_counters>
+
+class builtin_metrics : public utils::singleton<builtin_metrics>
```

Comment:
Is it possible to avoid using singleton?
https://www.davidtanzer.net/david's%20blog/2016/03/14/6-reasons-why-you-should-avoid-singletons.html

Reply:
OK, I'll use member instead of singleton. Since in most cases an instance is started as a meta server or a replica server, in fact there tends to be an object for `builtin_metrics`.